### PR TITLE
feat: index ad removal command

### PR DESCRIPTION
**Pull request by LexLuthr**
This command will send an ad to IPNI to remove the indexes for a particular deal. This command is to be used for removing deals not being cleaned up by automated system in Boost.

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -41,6 +41,8 @@ type Boost interface {
 	BoostDeal(ctx context.Context, dealUuid uuid.UUID) (*smtypes.ProviderDealState, error)                                                      //perm:admin
 	BoostDealBySignedProposalCid(ctx context.Context, proposalCid cid.Cid) (*smtypes.ProviderDealState, error)                                  //perm:admin
 	BoostDummyDeal(context.Context, smtypes.DealParams) (*ProviderDealRejectionInfo, error)                                                     //perm:admin
+	BoostIndexerAnnounceDealRemoved(ctx context.Context, pieceCid cid.Cid) (cid.Cid, error)                                                     //perm:admin
+	LegacyDealByProposalCid(ctx context.Context, propCid cid.Cid) (storagemarket.MinerDeal, error)                                              //perm:admin
```
Comment (nonsense) [inline api/api.go:45]: This is added under `Boost` MethodGroup, so maybe add `Boost` prefix?

File: api/proxy_gen.go [hunk 2]
```diff
@@ -454,6 +458,17 @@ func (s *BoostStub) BoostIndexerAnnounceAllDeals(p0 context.Context) error {
 	return ErrNotSupported
 }
 
+func (s *BoostStruct) BoostIndexerAnnounceDealRemoved(p0 context.Context, p1 cid.Cid) (cid.Cid, error) {
+	if s.Internal.BoostIndexerAnnounceDealRemoved == nil {
+		return *new(cid.Cid), ErrNotSupported
```
Comment (nonsense) [inline api/proxy_gen.go:463]: I think you can just use `cid.Undef` instead of `*new(cid.Cid)`

Comment (LexLuthr) [reply api/proxy_gen.go:463]: This is auto generated by the `make gen`. We might have to adjust the types in api gen for this. I am not sure if it is worth the required changes.

File: api/api.go (hunk 1 above)
Comment (LexLuthr) [reply api/api.go]: Updated. The reason to have it under Boost group is ensure compatibility after removing legacy markets.

**Review by nonsense — APPROVED** [review]

File: api/proxy_gen.go (hunk 2 above)
Comment (nonsense) [reply api/proxy_gen.go:463]: Yes, sorry, didn't notice this is the auto-generated part of the API.

**Pull request merged**